src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DataAnnotationsMetadataDetailsProvider.cs

```diff
@@ -0,0 +1,173 @@
+            {
+                context.BindingMetadata.IsReadOnly = !editableAttribute.AllowEdit;
+            }
+        }
+
+        /// <inheritdoc />
+        public void GetDisplayMetadata([NotNull] DisplayMetadataProviderContext context)
+        {
+            SetDisplayDetails(context.Attributes, context.DisplayMetadata);
+        }
+
+        private static void SetDisplayDetails(IReadOnlyList<object> attributes, DisplayMetadata details)
+        {
+            var dataTypeAttribute = attributes.OfType<DataTypeAttribute>().FirstOrDefault();
+            var displayAttribute = attributes.OfType<DisplayAttribute>().FirstOrDefault();
+            var displayColumnAttribute = attributes.OfType<DisplayColumnAttribute>().FirstOrDefault();
+            var displayFormatAttribute = attributes.OfType<DisplayFormatAttribute>().FirstOrDefault();
+            var hiddenInputAttribute = attributes.OfType<HiddenInputAttribute>().FirstOrDefault();
+            var scaffoldColumnAttribute = attributes.OfType<ScaffoldColumnAttribute>().FirstOrDefault();
+            var uiHintAttribute = attributes.OfType<UIHintAttribute>().FirstOrDefault();
+
+            // Special case the [DisplayFormat] attribute hanging off an applied [DataType] attribute. This property is
+            // non-null for DataType.Currency, DataType.Date, DataType.Time, and potentially custom [DataType]
+            // subclasses. The DataType.Currency, DataType.Date, and DataType.Time [DisplayFormat] attributes have a
+            // non-null DataFormatString and the DataType.Date and DataType.Time [DisplayFormat] attributes have
+            // ApplyFormatInEditMode==true.
+            if (displayFormatAttribute == null && dataTypeAttribute != null)
+            {
+                displayFormatAttribute = dataTypeAttribute.DisplayFormat;
+            }
+
+            // ConvertEmptyStringToNull
+            if (displayFormatAttribute != null)
+            {
+                details.ConvertEmptyStringToNull = displayFormatAttribute.ConvertEmptyStringToNull;
+            }
+
+            // DataTypeName
+            if (dataTypeAttribute != null)
+            {
+                details.DataTypeName = dataTypeAttribute.GetDataTypeName();
+            }
+            else if (displayFormatAttribute != null && !displayFormatAttribute.HtmlEncode)
+            {
+                details.DataTypeName = DataType.Html.ToString();
+            }
+
+            // Description
+            if (displayAttribute != null)
+            {
+                details.Description = displayAttribute.Description;
+            }
+
+            // DisplayFormat
+            if (displayFormatAttribute != null)
+            {
+                details.DisplayFormatString = displayFormatAttribute.DataFormatString;
+            }
+
+            // DisplayName
+            if (displayAttribute != null)
+            {
+                details.DisplayName = displayAttribute.Name;
```

Comment:
do not use the `Name` property.  instead call `displayAttribute.GetName()`.  (fyi these methods are how `[Display]` exposes values from resources)